examples/videotondi/videotondi.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	vidio "github.com/AlexEidt/Vidio"
+	"github.com/bitfocus/gondi"
+)
+
+var (
+	videoFrames int64 = 0
+)
+
+func videoToNDI(sender *gondi.SendInstance) {
+	for {
+		video, err := vidio.NewVideo("koala.mp4")
+		if err != nil {
+			log.Println("videoToNDI: failed to read video", err)
+			continue
+		}
+
+		for video.Read() {
+			startFrameTime := time.Now()
+			videoFrame := gondi.NewVideoFrameV2()
+			videoFrame.FourCC = gondi.FourCCTypeRGBA
+			videoFrame.FrameFormatType = gondi.FrameFormatProgressive
+			videoFrame.Xres = int32(video.Width())
+			videoFrame.Yres = int32(video.Height())
+			videoFrame.LineStride = 0 // 2 bytes per pixel
+			videoFrame.FrameRateN = 30000
+			videoFrame.FrameRateD = 1001
+			videoFrame.Data = &video.FrameBuffer()[0]
+			sender.SendVideoFrameAsync(videoFrame)
+
+			// force frame lock to the video FPS
+			// https://stackoverflow.com/a/61878644
```

Comment:
Since you have enabled video and audio blocking in the NDI sender instance, this shouldn't really be needed, as you could use the NDI library to do the timing, which it is very much better at, than this implementation :) Just use SendVideoFrame() instead of SendVideoFrameAsync()

Reply:
@haakonnessjoen ok makes sense will update :)